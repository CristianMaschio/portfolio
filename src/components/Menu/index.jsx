import React, { useContext } from "react";
import Context from "../../Context";
import { InfoCircleFilled } from "@ant-design/icons";

export default function Menu({ paragraphs }) {
  const { setRenderInformation } = useContext(Context);

  return (
    <div className="menu">
      <InfoCircleFilled className="info-menu" onClick={setRenderInformation} />
      <p onClick={setRenderInformation}>Cristian Maschio</p>
      <ul>
        {paragraphs?.map((paragraph) => (
          <li key={paragraph.id}>
            <a href={`#${paragraph.id}`}>{paragraph.title}</a>
          </li>
        ))}
      </ul>
    </div>
  );
}
