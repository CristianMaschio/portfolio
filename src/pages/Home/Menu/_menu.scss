.menu {
  position: sticky;
  background-color: $background-primary-color;
  top: 0;
  margin-left: 1px;
  color: white;
  height: 3rem;
  display: flex;
  z-index: 2;
  margin-top: -3rem;

  @media screen and (max-width: $mobile-max-width) {
    margin-left: 0px;
  }

  ul {
    flex: 1;
    display: flex;
    flex-direction: row;
    justify-content: center;
    list-style-type: none;
    li {
      margin-left: 1rem;
      margin-right: 1rem;
      a {
        color: white;
        text-decoration: none;
      }
    }
    @media screen and (max-width: $mobile-max-width) {
      display: none;
    }
  }

  .info-menu {
    display: none;
    cursor: pointer;
    font-size: 1.7rem;
    align-self: center;
    margin-left: 1rem;
    @media screen and (max-width: $mobile-max-width) {
      display: inline;
    }
  }

  p {
    display: none;
    cursor: pointer;
    margin-left: 0.5rem;
    @media screen and (max-width: $mobile-max-width) {
      display: inline;
    }
  }
}
