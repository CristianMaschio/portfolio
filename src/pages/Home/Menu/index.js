import React, { useContext } from "react"
import Context from "../../../Context"

import whiteInfo from "../../../assets/images/whiteInfo.png"
const menuLink = require(`../../../assets/languages/En-content.json`).menu
  
export default function Menu() {
  const { setRenderInformation } = useContext(Context)
  
  return (
    <div className="menu">
      <img
        onClick={setRenderInformation}
        src={whiteInfo}
        alt="info click icon"
        className="info-menu"
      />
      <p onClick={setRenderInformation}>Cristian Maschio</p>
      <ul>
        <li>
          <a href="#professional">{menuLink.professional} </a>
        </li>
        <li>
          <a href="#education">{menuLink.education}</a>
        </li>
        <li>
          <a href="#passions">{menuLink.passions}</a>
        </li>
        <li>
          <a href="#personal-experiences">{menuLink.experiences}</a>
        </li>
        <li>
          <a href="#code">{menuLink.code}</a>
        </li>
      </ul>
    </div>
  )
}
