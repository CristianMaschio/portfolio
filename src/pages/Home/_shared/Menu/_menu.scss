.menu {
  position: sticky;
  background-color: $background-primary-color;
  top: 0;
  margin-left: 1px;
  color: white;
  height: 3rem;
  display: flex;
  z-index: 2;

  @media screen and (max-width: $mobile-max-width) {
    margin-left: 0px;
  }

  ul {
    flex: 1;
    display: flex;
    flex-direction: row;
    justify-content: center;
    list-style-type: none;
    li {
      margin-left: 1rem;
      margin-right: 1rem;
      a {
        color: white;
        text-decoration: none;
      }
    }
    @media screen and (max-width: $mobile-max-width) {
      display: none;
    }
  }

  .info-menu {
    display: none;
    cursor: pointer;
    height: 1.5rem;
    align-self: center;
    margin-left: 1rem;
    @media screen and (max-width: $mobile-max-width) {
      display: inline;
    }
  }

  p {
    display: none;
    cursor: pointer;
    margin-left: 0.5rem;
    @media screen and (max-width: $mobile-max-width) {
      display: inline;
    }
  }

  .language-container {
    width: 4rem;
    display: flex;
    flex-direction: row;
    align-self: center;
    color: white;
    @media screen and (max-width: $mobile-max-width) {
      flex: 1;
      padding-right: 1rem;
      justify-content: flex-end;
    }
    p {
      display: inline;
      margin: 0.2rem;
      font-weight: 100;
      cursor: pointer;
    }
    .selected-language {
      font-weight: bold;
    }
  }
}
