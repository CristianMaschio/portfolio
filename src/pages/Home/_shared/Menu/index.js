import React, { PureComponent } from "react"
import whiteInfo from "../../../../assets/images/whiteInfo.png"

import AppState from "../../../../gState/context"

export class Menu extends PureComponent {
  render() {
    const { setRenderInformation } = this.props.context
    const menuLink = require(`../../../../assets/languages/En-content.json`)
      .menu
    return (
      <div className="menu">
        <img
          onClick={setRenderInformation}
          src={whiteInfo}
          alt="info click icon"
          className="info-menu"
        />
        <p onClick={setRenderInformation}>Cristian Maschio</p>
        <ul>
          <li>
            <a href="#professional">{menuLink.professional} </a>
          </li>
          <li>
            <a href="#education">{menuLink.education}</a>
          </li>
          <li>
            <a href="#passions">{menuLink.passions}</a>
          </li>
          <li>
            <a href="#personal-experiences">{menuLink.experiences}</a>
          </li>
          <li>
            <a href="#code">{menuLink.code}</a>
          </li>
        </ul>
      </div>
    )
  }
}

export default class MenuConsumer extends PureComponent {
  render() {
    return (
      <AppState.Consumer>{state => <Menu context={state} />}</AppState.Consumer>
    )
  }
}
