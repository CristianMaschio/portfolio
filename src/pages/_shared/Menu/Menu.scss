@import "../../../styles/colors";

.menu {
  position: sticky;
  background-color: $dark-color;
  top: 0;
  margin-left: 1px;
  color: white;
  height: 3rem;
  display: flex;
  align-items: center;
}
