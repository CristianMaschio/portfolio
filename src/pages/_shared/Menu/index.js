import React, { PureComponent } from "react"
import "./Menu.scss"
import whiteInfo from "./whiteInfo.png"

import AppState from "../../../gState/context"

export class Menu extends PureComponent {
  renderLanguageSelector() {
    const { language, changeLanguage } = this.props.context
    const isEn = language === "En"

    return (
      <div className="language-container">
        <p
          className={isEn ? "selected-language" : ""}
          onClick={() => !isEn && changeLanguage("En")}
        >
          En
        </p>
        <p>-</p>
        <p
          className={!isEn ? "selected-language" : ""}
          onClick={() => isEn && changeLanguage("It")}
        >
          It
        </p>
      </div>
    )
  }
  render() {
    const { setRenderInformation, language } = this.props.context
    const menuLink = require(`../../../data/${language}-content.json`).menu
    return (
      <div className="menu">
        <img
          onClick={setRenderInformation}
          src={whiteInfo}
          className="info-menu"
        />
        <p onClick={setRenderInformation}>Cristian Maschio</p>
        <ul>
          <li>
            <a href="#professional">{menuLink.professional} </a>
          </li>
          <li>
            <a href="#education">{menuLink.education}</a>
          </li>
          <li>
            <a href="#passions">{menuLink.passions}</a>
          </li>
          <li>
            <a href="#personal-experiences">{menuLink.experiences}</a>
          </li>
        </ul>{" "}
        {this.renderLanguageSelector()}
      </div>
    )
  }
}

export default class MenuConsumer extends PureComponent {
  render() {
    return (
      <AppState.Consumer>{state => <Menu context={state} />}</AppState.Consumer>
    )
  }
}
