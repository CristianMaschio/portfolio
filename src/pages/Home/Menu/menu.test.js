import React from "react"
import renderer from "react-test-renderer"
import Menu from "."

it("Menu: render correctly", () => {
  const menu = renderer
    .create(
      <Menu
        paragraphs={[
          { id: "test", title: "test" },
          { id: "test1", title: "test1" },
        ]}
      />
    )
    .toJSON()

  expect(menu).toMatchSnapshot()
})
