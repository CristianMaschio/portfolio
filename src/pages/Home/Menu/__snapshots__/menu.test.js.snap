// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu: render correctly 1`] = `
<div
  className="menu"
>
  <span
    aria-label="info-circle"
    className="anticon anticon-info-circle info-menu"
    onClick={[Function]}
    role="img"
    tabIndex={-1}
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="info-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
      />
    </svg>
  </span>
  <p
    onClick={[Function]}
  >
    Cristian Maschio
  </p>
  <ul>
    <li>
      <a
        href="#test"
      >
        test
      </a>
    </li>
    <li>
      <a
        href="#test1"
      >
        test1
      </a>
    </li>
  </ul>
</div>
`;
